daft/delta_lake/delta_lake_scan.py

```diff
@@ -43,6 +44,22 @@ def __init__(
         deltalake_sdk_io_config = storage_config.config.io_config
         scheme = urlparse(table_uri).scheme
         if scheme == "s3" or scheme == "s3a":
+            # Try to get region from boto3
+            if deltalake_sdk_io_config.s3.region_name is None:
+                from botocore.exceptions import BotoCoreError
+
+                try:
+                    client = boto3_client_from_s3_config("s3", deltalake_sdk_io_config.s3)
+                    response = client.get_bucket_location(Bucket=urlparse(table_uri).netloc)
+                except BotoCoreError:
+                    # If bucket location request fails, ignore error because S3 config may be correctly populated later from environment
```

Comment:
We should log rather than failing silently. Check out logger.exception for the canonical way of doing this

Reply:
I didn't really want to log anything here because it could be confusing to users. This is actually why I put this before the code that gets the config from the environment, because that part logs when it fails and I have seen users report this in the past when the actual root issue was elsewhere.

What if I log it as warning with a friendly message?

Reply:
Ok I logged an informative warning instead